rust/windows-client/src/main.rs

```diff
@@ -0,0 +1,105 @@
+// Sample code from https://github.com/microsoft/windows-rs/blob/0.52.0/crates/samples/windows/create_window/src/main.rs
+
+use windows::{
+    core::*,
+    Win32::{
+        Foundation::*,
+        Graphics::Gdi::ValidateRect,
+        System::LibraryLoader::GetModuleHandleA,
+        UI::{Shell::*, WindowsAndMessaging::*},
+    },
+};
+
+fn main() -> Result<()> {
+    unsafe {
+        let instance = GetModuleHandleA(None)?;
+        debug_assert!(instance.0 != 0);
+
+        let window_class = s!("window");
+
+        let wc = WNDCLASSA {
+            hCursor: LoadCursorW(None, IDC_ARROW)?,
+            hInstance: instance.into(),
+            lpszClassName: window_class,
+
+            style: CS_HREDRAW | CS_VREDRAW,
+            lpfnWndProc: Some(wndproc),
+            ..Default::default()
+        };
+
+        let atom = RegisterClassA(&wc);
+        debug_assert!(atom != 0);
+
+        let window = CreateWindowExA(
+            WINDOW_EX_STYLE::default(),
+            window_class,
+            s!("This is a sample window"),
+            WS_OVERLAPPEDWINDOW | WS_VISIBLE,
+            CW_USEDEFAULT,
+            CW_USEDEFAULT,
+            CW_USEDEFAULT,
+            CW_USEDEFAULT,
+            None,
+            None,
+            instance,
+            None,
+        );
+
+        // Create tray icon
+
+        let sz_tip = {
+            let mut bytes = [0u8; 128];
+            let s = "Firezone client\0";
+            bytes[0..s.len()].copy_from_slice(s.as_bytes());
+            bytes
+        };
+
+        // let icon = LoadIconA(instance, PCSTR ("1\0".as_bytes().as_ptr())).unwrap();
```

Comment:
No, just happened to be fighting the LoadIconA function when I committed this one. In my local repo I'm now trying NWG and Tauri so this should be gone next time. I couldn't get LoadIcon to work.

Reply:
Ah, let's remove it before merging it then.

Also just as a note, I think we tried out tauri before and we decided not to go with it, at least for the other native platforms, I don't exactly remember the reasons maybe @jamilbk remembers.

Reply:
He said earlier in DM "our reasoning for not going with Tauri was because the it was too much overhead vs just using platform idiomatic approaches on Apple"

I'm trying Tauri now and yeah the RAM usage to do nothing is like 100 MB. But the API is very nice. Up to you two.

Reply:
@conectado Since I'm just trying to get the CI to produce an exe, could I just cut it down to a "Hello World" CLI? That way the Rust code doesn't slow down this PR. Once the CI/CD is working I can put it back in.

Reply:
"too much overhead" was more in reference to engineering overhead (security, troubleshooting, cognitive, maintainability, etc)

Reply:
I suppose what I fear is architecting the app around Tauri then finding out later it doesn't work well for low-level interfacing or something and then having to unwind the app architecture

Reply:
But I guess at the end of the day it's all just Rust and the UI layer is pretty thin, so we should be able to refactor to use other UI libs when we get further along. Probably better to go with whatever's fastest easiest for you @ReactorScram since maintainability isn't a worry for some time